File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque

def BFS(coms, idx, visited):
    queue = deque([])
    queue.append(idx)
    visited[idx] = 1
    while queue:
        target = queue.popleft()
        for idx in coms[target]:
            if visited[idx] == 0:
                queue.append(idx)
                visited[idx] = 1
            
def solution(n, computers):
    answer = 0
    visited = [0 for i in range(n)]
    coms = {}
    for idx in range(n):
        coms[idx] = [i for (i, elem) in enumerate(computers[idx]) if i != idx and elem == 1]
    for (idx, elem) in coms.items():
        if visited[idx] == 0:
            BFS(coms, idx, visited)
            answer += 1
    return answer

File: 프로그래머스/lv3/43162. 네트워크/네트워크.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var visited: [Bool] = Array(repeating: false, count: n)
    var result = 0
    func BFS(startIdx: Int) {
        var queue: [Int] = [startIdx]
        visited[startIdx] = true
        while !queue.isEmpty {
            let target = queue.removeFirst()

            for idx in 0..<computers[target].count {
                if idx == target {
                    continue
                }
                if computers[target][idx] == 1 && !visited[idx] {
                    queue.append(idx)
                    visited[idx] = true
                }
            }
        }
    }
    
    for i in 0..<n {
        if !visited[i] {
            BFS(startIdx: i)
            result += 1
        }
    }
    return result
}